### PR TITLE
[pickers] Remove the `PaperContent` slot from the new pickers

### LINKED ISSUE #7278: [pickers] Remove `PaperContent` slot
After the introduction of [Layout](https://github.com/mui/mui-x/pull/6900) it no longer makes sense in having [PaperContent slot](https://next.mui.com/x/react-date-pickers/custom-components/#paper-content), which is more limited (does not apply to mobile pickers) and allows more limited customization options.

### PR DESCRIPTION
**Pull request by flaviendelangle**
Fixes #7278

### CODE REVIEW

File: docs/data/date-pickers/custom-layout/custom-layout.md [hunk 1]
```diff
@@ -72,18 +72,20 @@ Use the `usePickerLayout` hook to get the subcomponents React nodes.
 Then you can fully customize the DOM structure.
 
 ```jsx
-const { usePickerLayout } from '@mui/x-date-pickers/PickersLayout';
+import { usePickerLayout } from '@mui/x-date-pickers/PickersLayout';
```
Comment (LukasTy) [inline docs/data/date-pickers/custom-layout/custom-layout.md:75]:
```suggestion
import { usePickerLayout, PickersLayoutRoot } from '@mui/x-date-pickers/PickersLayout';
```

File: docs/data/date-pickers/custom-layout/custom-layout.md [hunk 2]
```diff
@@ -72,18 +72,20 @@ Use the `usePickerLayout` hook to get the subcomponents React nodes.
...
-    {toolbar}
-    {actionBar}
-    {content}
-  </div>
-}
+  return (
+    <PickersLayoutRoot>
+      {toolbar}
+      {actionBar}
+      {content}
```
Comment (LukasTy) [inline docs/data/date-pickers/custom-layout/custom-layout.md:85]: In a final example use would have to use `PickersLayoutContentWrapper` to wrap the content, but for the brevity of showing the idea it's probably better to use as minimal example as possible. 👌

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md [hunk 3]
```diff
@@ -476,24 +476,67 @@ The `locale` prop of the `LocalizationProvider` component have been renamed `ada
...
-- The component slot `LeftArrowButton` has been renamed `PreviousIconButton` on all pickers:
+- The `PaperContent` / `paperContent` component slot and component props slot have been removed.
 
-  ```diff
-   <DatePicker
-     components={{
-  -    LeftArrowButton: CustomButton,
-  +    PreviousIconButton: CustomButton,
-     }}
+  You can use the new `Layout` component slot.
```
Comment (LukasTy) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:483]:
```suggestion
  You can use the new [`Layout` component slot](/x/react-date-pickers/custom-layout/).
```

**Review by LukasTy — APPROVED** [review]

File: docs/data/date-pickers/custom-layout/custom-layout.md (hunk 2 above)
Comment (flaviendelangle) [reply docs/data/date-pickers/custom-layout/custom-layout.md:85]: I don't know which one is better here
I fear that a misleading code snippet lead people to build broken apps :/

Comment (LukasTy) [reply docs/data/date-pickers/custom-layout/custom-layout.md:88]: Yes, that is my concern as well, especially when the longer example has the code collapsed.

Comment (flaviendelangle) [reply docs/data/date-pickers/custom-layout/custom-layout.md:88]: Do you prefer to keep as it is on the PR or to add `PickersLayoutContentWrapper` ?

Comment (LukasTy) [reply docs/data/date-pickers/custom-layout/custom-layout.md:88]: I'd vote for including it just for the sake of someone not expanding the complete demo and later complaining about having issues with using layout slot. 🙈

Comment (flaviendelangle) [reply docs/data/date-pickers/custom-layout/custom-layout.md]: Done

File: docs/data/date-pickers/custom-layout/custom-layout.md [hunk 4]
```diff
@@ -72,18 +72,27 @@ Use the `usePickerLayout` hook to get the subcomponents React nodes.
 Then you can fully customize the DOM structure.
 
 ```jsx
-const { usePickerLayout } from '@mui/x-date-pickers/PickersLayout';
+import {
+  usePickerLayout,
+  PickersLayoutRoot,
+  pickersLayoutClasses,
```
Comment (LukasTy) [inline docs/data/date-pickers/custom-layout/custom-layout.md:78]:
```suggestion
  pickersLayoutClasses,
  PickersLayoutContentWrapper,
```
🙈

**Pull request merged**